Debugging botio on Windows

Rob--W opened the pull request:
DO NOT MERGE. This is just to debug our windows bot, which often fails (see e.g. http://107.22.172.223:8877/e5449911cdcb677/output.txt).

/cc @timvandermeij

timvandermeij commented:
/botio-windows unittest

timvandermeij commented:
/botio-windows unittest

timvandermeij commented:
/botio-windows unittest

timvandermeij commented:
/botio-windows test

timvandermeij commented:
/botio-windows unittest

brendandahl commented:
I'm not sure if you're aware of the context, but the bot fails when there is a firefox upgrade on windows and it does not close on the unit tests. I think it may be changing process id's on the upgrade run.

Rob--W committed:
Improve robustness of builder (esp. on Windows)

- Use rimraf instead of a custom removeDirSync implementation - rimraf
  deals with edge cases like EPERM on Windows.

- Detect when the process exits before it was requested via stop(),
  instead of running the cleanup handler.

- Add fallback for process detection when the process exits before it
  was requested. On *nix systems, this is done via pkill and pgrep, on
  Windows this is done via wmic.

- Add some asserts to check the preconditions of the methods, and output
  some status information to aid debugging in case of failure.

I have verified that these changes work on ArchLinux and Windows XP,
using Chrome and Firefox, as follows:

1. node make unittest

2. node make unittest
3. Restart the Firefox process via the task manager as soon as possible.

4. node make unittest
5. Temporary lock a file/directory within the temporary profile
   directory until the tests have finished, and then unlock the file
   within 10 seconds.

In all cases, the auxilary browser processes are killed, and the
temporary profile directory is wiped.

Rob--W committed:
Disable the regular kill-by-pid case

To allow botio to test the kill-by-pgrep/pkill/wmic case.

timvandermeij commented:
/botio-windows test

Rob--W committed:
Reduce verbosity of output in test/webbrowser.js

timvandermeij commented:
/botio-windows unittest
